lib/portlayer/exec/exec.go

```diff
@@ -171,6 +182,147 @@ func eventCallback(ie events.Event) {
 	return
 }
 
+func waitForContainerStarted(container *Container) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), propertyCollectorTimeout)
+	defer cancel()
+
+	if container.vm == nil {
+		log.Errorf("Container(%s) wait failed for vm is not found", container.ExecConfig.ID)
+		return false
+	}
+	log.Debugf("Waiting for container(%s) started", container.ExecConfig.ID)
+	for _, k := range extraconfig.CalculateKeys(container.ExecConfig, "Sessions.*.Detail.StartTime", "") {
+		var poweredoff bool
+		var detail string
+		waitFunc := func(pc []types.PropertyChange) bool {
```

Comment:
Don't like this being copied here from vm.go.

If you could make this function block a var, that can be used by both vm.go and here, you can wrap it in another lambda that will just do the check for `detail != "0"` after the wrapped function (the new var) returns true.